java-checks/src/test/java/org/sonar/java/checks/maven/helpers/MavenDependencyVersionMatcherTest.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.maven.helpers;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.sonar.maven.model.LocatedAttribute;
+import org.sonar.maven.model.maven2.Dependency;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class MavenDependencyVersionMatcherTest {
```

Comment:
This test class is really long. 
I didn't investigate in deep but I'm pretty sure there are a lot of redundant tests.
Moreover : you can factorize things : 
```
setDependency(StringVar); 
assertThat(matcher.matches(dependency)).isTrue();
```
matcher can be made into a field and you can then declare assertMatch and assertNotMatch methods that would take a string parameter.